tests/test_core_attention.py

```diff
@@ -109,3 +109,52 @@ def test_amp_attention_sparsecs(device):
 
     expected_device = torch.float32
     assert r.dtype == expected_device
+
+
+@pytest.mark.parametrize("device", ["cuda"])
+@pytest.mark.parametrize("data_type", [torch.float16, torch.float32])
+def test_switch_blocksparse(device, data_type):
```

Comment:
An additional test for when sequence length is not divisible by block size could be useful